tests/test_mathematical.py

```diff
@@ -1289,6 +1289,58 @@ def test_op_multiple_dtypes(dtype1, func, dtype2, data):
         assert_allclose(result, expected)
 
 
+class TestI0:
```

Comment:
For the special case of `dpnp.inf`, it does not return correct result.

```python
import numpy, cupy, dpnp

numpy.i0(numpy.inf)
#  array(nan)  # although the correct result is inf

cupy.i0(cupy.inf)
#  array(inf)

dpnp.i0(dpnp.array(dpnp.inf))
# array(1.)

Reply:
Implementation is updated, tests are added to cover that.